packages/core/src/fhir-deduplication/__tests__/group-same-observations.test.ts

```diff
@@ -104,6 +104,49 @@ describe("groupSameObservationsSocial", () => {
     expect(observationsMap.size).toBe(2);
   });
 
+  it("does not group observations with unknown codes and different displays/texts", () => {
```

Comment:
nit: it would be good to have a single validation being done on each test; in this case, I can't tell if the code works because:
1. `code.coding[1].display` is different; or
2. `code.text` is different; or
3. both are different.